src/cmd/factory.go

```diff
@@ -76,6 +79,27 @@ func getLifecycle() *lifecycle.Lifecycle {
 	return services.lifecycle
 }
 
+// getEventListener creates or returns existing event listener service.
+func getEventListener(cfg *config.Config) event.Listener {
+	if services.eventListener == nil {
+		services.eventListener = event.NewListener(
+			getEventManager(cfg),
+			getInclusionReportSentEventHandler(cfg),
+		)
+	}
+
+	return services.eventListener
+}
+
+// getInclusionReportSentEventHandler creates or returns existing schedule listener.
+func getInclusionReportSentEventHandler(cfg *config.Config) *event.Handler {
```

Comment:
I think there's no need to store the handler in a DI container. Please create it directly in an event listener factory.